cpp/src/parquet/column_reader.cc

```diff
@@ -822,6 +822,7 @@ class ColumnReaderImplBase {
       max_size -= def_levels_bytes;
     }
 
+    current_page_may_have_nulls_ = max_def_level_ > 0 || max_rep_level_ > 0;
```

Comment:
maybe a better name or comment for `nulls_` ? Since `max_rep_level_ > 0` and null is weird